lib/ProofSet.js

```diff
@@ -306,9 +306,10 @@ async function _verify({
   document, suites, proofSet, purpose,
   documentLoader, expansionMap, compactProof}) {
   // filter out matching proofs
-  const matches = proofSet.filter(proof => purpose ?
+  const result = await Promise.all(proofSet.map(proof => purpose ?
     purpose.match(proof, {document, documentLoader, expansionMap}) :
-    true);
+    true));
+  const matches = proofSet.filter((value, index) => result[index]);
```

Comment:
Why are we using `index`? Does the order change somehow?

Reply:
we are filtering `proofSet` based on `result`.  Is there a better way?

Reply:
Hmm, I guess it's fine. I was thinking we would map to the `proof` or `null` based on the result of awaiting `match` during the `map()` call and then filter out based on `null` or not (like below). But I suppose this gets the job done fine.